src/render/shape.cpp

```diff
@@ -154,14 +155,16 @@ void embree_intersect_scalar(int* valid,
 #endif
         }
     } else {
-        if (dr::all(shape->ray_test(ray)))
+        if (dr::all(shape->ray_test(ray, true, primID)))
```

Comment:
Same here

Reply:
Thanks so much! I've fixed both these directly in the `sdf` branch